Firestore/Example/Tests/Integration/API/FIRValidationTests.mm

```diff
@@ -533,6 +544,11 @@ - (void)testQueryInequalityFieldMustMatchFirstOrderByField {
   XCTAssertNoThrow([[[coll queryOrderedByField:@"x"] queryOrderedByField:@"y"] queryWhereField:@"x"
                                                                                  isGreaterThan:@32],
                    @"inequality same as first order by works.");
+
+  XCTAssertNoThrow([[coll queryOrderedByField:@"x"] queryWhereField:@"y" isEqualTo:@"cat"],
+                   @"equality different than orderBy works.");
+  XCTAssertNoThrow([[coll queryOrderedByField:@"x"] queryWhereField:@"y" arrayContains:@"cat"],
+                   @"array_contains different than orderBy works.");
```

Comment:
Perhaps verify that multiple arrayContains (on distinct fields) are allowed in the same query?

Reply:
Good idea, though I added this as an actual integration test in FIRQueryTests.mm rather than here.